master/buildbot/status/web/hooks/github.py

```diff
@@ -168,6 +184,73 @@ def _process_change(self, payload, user, repo, repo_url, project):
 
         return changes
 
+    def _process_pull_request(self, payload, user, repo, repo_url, project, codebase=None):
+        changes = []
+        number = payload['number']
+
+        # We only care about opened/reopened/synchronize pull requests
+        if payload['action'] not in ['opened', 'reopened', 'synchronize']:
+            log.msg("Ignoring `%s': Not an opened pull request" % number)
+            return changes
+
+        # refname = payload['pull_request']['head']['ref']
+        branch = 'refs/pull/%s/head' % number
+        category = 'pull-request'
+
+        if payload['pull_request']['mergeable'] == False:
+            log.msg("Pull request `%s' not mergeable, ignoring" % branch)
+            return changes
+
+        r = requests.get(payload['pull_request']['commits_url'] + "?per_page=100")
```

Comment:
Also needs to use txrequests. This might mean change the whole hook framework to be asynchronous